core/src/main/java/cz/xtf/core/waiting/failfast/SupplierFailFastCheck.java

```diff
@@ -0,0 +1,45 @@
+package cz.xtf.core.waiting.failfast;
+
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Generic class for {@link FailFastCheck} so that builder can build it easily.
+ */
+public class SupplierFailFastCheck<X> implements FailFastCheck {
+
+	private final Supplier<X> eventSupplier;
+	private final Function<X, Boolean> shouldFail;
+	private final Function<X, String> reasonSupplier;
+
+	private String reason;
+
+	/**
+	 * @param supplier       supplier that provide object
+	 * @param shouldFail     condition. The result is the result of {@link FailFastCheck#shouldFail()}
+	 * @param reasonSupplier returns string as a reason why {@code shouldFail} returned {@code true}
+	 */
+	SupplierFailFastCheck(Supplier<X> supplier, Function<X, Boolean> shouldFail, Function<X, String> reasonSupplier) {
```

Comment:
`supplier` is somewhat generic name of param and javadoc does not help much what it is. What about `watchedResourcesSupplier`

Reply:
Yeah, sounds better.